http/src/v0/refs.rs

```diff
@@ -0,0 +1,76 @@
+use futures::stream;
+use ipfs::error::Error;
+use ipfs::{Ipfs, IpfsTypes};
+use serde::Serialize;
+use warp::hyper::Body;
+use warp::{path, Filter, Rejection, Reply};
+
+use super::support::with_ipfs;
+
+#[derive(Serialize, Debug)]
+struct RefsResponseItem {
+    #[serde(rename = "Err")]
+    err: String,
+
+    #[serde(rename = "Ref")]
+    refs: String,
+}
+
+/// Handling of https://docs-beta.ipfs.io/reference/http/api/#api-v0-refs-local
+pub fn local<T: IpfsTypes>(
+    ipfs: &Ipfs<T>,
+) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
+    path!("refs" / "local")
+        .and(with_ipfs(ipfs))
+        .and_then(inner_local)
+}
+
+async fn inner_local<T: IpfsTypes>(ipfs: Ipfs<T>) -> Result<impl Reply, Rejection> {
+    let refs: Vec<Result<String, Error>> = ipfs
+        .refs_local()
+        .await
+        .unwrap()
+        .iter()
+        // TODO: Error handling
+        // .map_err(|err| RefsResponseItem { refs: "", err: "" })
+        .map(|cid| cid.as_ref().unwrap().to_string())
+        .map(|refs| RefsResponseItem {
+            refs,
+            err: "".to_string(),
```

Comment:
the `"".to_string()` means "allocate a new string pls". I can fix this when reconciling my `refs` implementation, but while we are not really in a position to avoid all allocations, I think we should avoid the easy allocations.

This could not have the allocation if the field type was `Cow<'static, str>` which means "either a &'static str or a String", and here a `&'static str` of `""` would do just fine.

Reply:
I'll handle this, resolving.